File: 94.java
// 94. Binary Tree Inorder Traversal
// https://leetcode.com/problems/binary-tree-inorder-traversal/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    List<Integer> out = new LinkedList();
    public List<Integer> inorderTraversal(TreeNode root) {
        traverse(root);
        return out;
    }
    private void traverse(TreeNode root) {
        if (root == null) {
            return;
        }
        
        traverse(root.left);
        out.add(root.val);
        traverse(root.right);
    }
}

File: 94.js
// 94. Binary Tree Inorder Traversal
// https://leetcode.com/problems/binary-tree-inorder-traversal/

/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root, output = []) {
    // L N R
    if (root === null) {
        return output
    }
    if (root.left) {
        inorderTraversal(root.left, output)
    }
    output.push(root.val)
    if (root.right) {
        inorderTraversal(root.right, output)
    }
    return output
};